src/plugins/plot/legend/PlotLegendItemExpanded.vue

```diff
@@ -144,40 +141,59 @@ export default {
     }
   },
   mounted() {
+    this.seriesModels = [];
     eventHelpers.extend(this);
     this.config = this.getConfig();
+    this.listenTo(this.config.series, 'add', this.onSeriesAdd, this);
+    this.listenTo(this.config.series, 'remove', this.onSeriesRemove, this);
+    this.config.series.forEach(this.onSeriesAdd, this);
     this.legend = this.config.legend;
     this.loaded = true;
-    this.listenTo(
-      this.seriesObject,
-      'change:color',
-      (newColor) => {
-        this.updateColor(newColor);
-      },
-      this
-    );
-    this.listenTo(
-      this.seriesObject,
-      'change:name',
-      () => {
-        this.updateName();
-      },
-      this
-    );
-    this.subscribeToStaleness(this.seriesObject.domainObject);
-    this.initialize();
   },
   beforeUnmount() {
     this.stopListening();
   },
   methods: {
+    onSeriesAdd(series, index) {
+      this.seriesModels[index] = series;
+      if (series.keyString === this.seriesKeyString) {
+        this.listenTo(
+          series,
+          'change:color',
+          (newColor) => {
+            this.updateColor(newColor);
+          },
+          this
+        );
+        this.listenTo(
+          series,
+          'change:name',
+          () => {
+            this.updateName();
+          },
+          this
+        );
+        this.subscribeToStaleness(series.domainObject);
+        this.initialize();
+      }
+    },
+    onSeriesRemove(series, index) {
+      this.seriesModels.splice(index, 1);
```

Comment:
Consider using the following approach instead of using the `index` to remove the series. Reordering series can change the indices of series and this could end up in removing the wrong item.
https://github.com/nasa/openmct/pull/6961/files#diff-ff5740a785ad44ad9fcb7df262125bf8857a763020881fe65802a8aae49da8aaR205